### PR TITLE
Go to 4 weeks

### PR DESCRIPTION
**Pull request by robvk**
Biggest changes:

- Moved all reading materials to the study book so other HYF chapters can also use it!
- Added Week4 which is fully about mongodb
- Updated prep exercises so that they are mentioned in the MAKEME
- Updated Week3 mongodb exercise

### CODE REVIEW

**Review by GinaSt — APPROVED** [review]

File: Week1/README.md [hunk 1]
```diff
@@ -4,14 +4,9 @@
...
 
-1. What is an information (system)?
-2. What are entities?
-3. What is a database?
-4. What is the role of a database in an application?
-5. What is Structured Query Language (SQL)?
-6. What are data types (as applied to databases)?
-7. How to use SQL to Create, Read, Update and Delete (CRUD)
-8. What is a database dump?
+1. [Introduction to databases](https://hackyourfuture.github.io/study/#/databases/README)
```
Comment (unmeshvrije) [inline Week1/README.md:7]: IMHO, it would be better to put videos from the same person/channel (preferably without a difficult-to-understand English accent); That way our students can follow them better

File: Week1/README.md [hunk 2]
```diff
@@ -21,187 +16,13 @@ These are the topics for week 1:
...
 
-Your teachers Igor and Unmesh have made video lectures for this week's material. You can find them here: [Videos 1 - 8](https://www.youtube.com/playlist?list=PLVYDhqbgYpYVq-rHFB1x2-P7esnWotCB7)
+This week we want to introduce you to the world of databases! We will start with a general introduction to what databases are, going over some terms and how to start thinking about your application's data. Have a look for the introduction [here](https://hackyourfuture.github.io/study/#/databases/README).
 
-<a href="https://www.youtube.com/playlist?list=PLVYDhqbgYpYVq-rHFB1x2-P7esnWotCB7" target="_blank"><img src="../assets/playlist-thumbnail.png" width="600" height="350" alt="HYF Video" /></a>
+In this module we will be covering both the relational type of databases (SQL) as well as document based databases (NoSQL). For now you do not need to know the difference, that will become apparent when we have covered both. We will start by going into relational database by learning about SQL, which is the language used when interacting with most relational databases. Have a look [here](https://hackyourfuture.github.io/study/#/databases/SQL).
 
-## 1. What is an information (system)?
```
Comment (unmeshvrije) [inline Week1/README.md:30]: Removing all these contents make `LESSONPLAN` and `README` out of sync

File: Week2/README.md [hunk 3]
```diff
@@ -4,305 +4,23 @@
...
-3. What are indexes?
-4. What’s domain modeling?
-   - Entity Relationship Diagram (ERD)
-5. How to use SQL: joins, aggregate Functions, distinct, group by, having
-   - Joins
-   - Aggregate functions
-   - Distinct
-   - Group By
-   - Having
+1. [Identifiers (Primary key, Foreign key, Composite key)](https://hackyourfuture.github.io/study/#/databases/sql/identifiers)
```
Comment (unmeshvrije) [inline Week2/README.md:7]: These topics align have the corresponding material in the study material! Nice 🎉

File: Week3/MAKEME.md [hunk 4]
```diff
@@ -3,21 +3,33 @@
...
-these exercises are NOT covered in the reading material. This is on purpose.
+- [MongoDB practice exercises](https://gist.githubusercontent.com/theRemix/7305403e1ab6fc8674f0/raw/c068ab51e930eb133a9443caa314205a89ef4d61/exercise.md) Note that this is a raw markdown file so you may want to copy it into your editor and save it as `.md` to get syntax highlighting. The live version has comments with answers which you can check out after over [here](https://gist.github.com/theRemix/7305403e1ab6fc8674f0#file-exercise-md).
 
-**Exercise 1 : Normalization**
+## 2. **Prep exercises**
+
+> Prep exercises are exercises that you should work on _before_ the session on Sunday. These are a little more difficult or show an important concept and as such are a great exercise to talk about with your mentor. Have a solution ready by Sunday as you may be asked to show what you did.
+
+Inside your `databases` fork, go to the folder `Week3`. Inside of that folder, navigate to `QA_PREP_EXERCISE.md` file which explains what needs to be done. There will also be some questions at the bottom to think about. Go through them _before_ the session on Sunday as it will be covered then.
```
Comment (unmeshvrije) [inline Week3/MAKEME.md:24]: | Go through them _before_ the session on Sunday as it will be covered then.
-> Some students might think that it will be covered anyway, so why study. Instead we could phrase it as
` Answer the questions at the end and send them to your teacher before the class`

File: Week1/README.md [hunk 5]
```diff
@@ -4,14 +4,9 @@
...
-1. What is an information (system)?
-2. What are entities?
-3. What is a database?
-4. What is the role of a database in an application?
-5. What is Structured Query Language (SQL)?
-6. What are data types (as applied to databases)?
-7. How to use SQL to Create, Read, Update and Delete (CRUD)
-8. What is a database dump?
+1. [Introduction to databases](https://hackyourfuture.github.io/study/#/databases/README)
+2. [Introduction to Structured Query Language (SQL)](https://hackyourfuture.github.io/study/#/databases/SQL)
```
Comment (unmeshvrije) [inline Week1/README.md:8]: This link gives 404, correct link is `https://hackyourfuture.github.io/study/#/databases/sql/README`

File: Week1/README.md [hunk 6]
```diff
@@ -4,14 +4,9 @@
...
+3. [Using database dumps](https://hackyourfuture.github.io/study/#/databases/dumps)
```
Comment (unmeshvrije) [inline Week1/README.md:9]: This link also gives 404

A suggestion to check all links by copy-pasting the URLs into a browser.

File: Week4/README.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# Reading Material Databases Week 4
+
+## Agenda
+
+These are the topics for week 4, all in MongoDB:
+
+- [Embedded vs Normalised data modeling](https://hackyourfuture.github.io/study/#/databases/nosql/mongodb/embedded-vs-normalised.md)
```
Comment (unmeshvrije) [inline Week4/README.md:7]: Do we need a `LESSONPLAN.md` for week 4 ?
If I look at the study material forNoSQL and Mongo, it is hard to follow with the example of `collection_name` 

We follow the example of `employee-departments-projects` tables through out the lesson plans and I would continue creating MongoDB `documents/collections` for one chosen database (be it a movie database or projects database)

**Review by unmeshvrije — CHANGES_REQUESTED** [review]
Let us discuss the LESSONPLAN and study material related comments

File: Week1/README.md (hunk 2 above)
Comment (robvk) [reply Week1/README.md:30]: Good point, resynced them by providing subtopics where needed

File: Week3/MAKEME.md (hunk 4 above)
Comment (robvk) [reply Week3/MAKEME.md:24]: Talked with Tjebbe about this, we feel this is a personal responsibility of the students to do it. If they don't they will fail at some point and we are confident we will catch when they don't understand the materials. We also don't want to put extra pressure on our Q&A mentors to do the checking for this

File: Week4/README.md (hunk 7 above)
Comment (robvk) [reply Week4/README.md:7]: Will do this in a new PR before week 4 starts. Good point, I forgot about it
